### DISCUSSION TITLE
In DataGrid, CheckBox.IsChecked is inconsistent with binding value

### DISCUSSION THREAD

**Discussion by JHING** [AvaloniaUI/Avalonia#17228] [Q&A]
### Describe the bug

In DataGrid, when an item is removed, a new item is added at the bottom; the checkbox.IsChecked of the new item is inconsistent with the binding value;
But when I set the checkbox's IsChecked before removing the item, this problem does not occur.

### To Reproduce

Here is the code to reproduce: https://github.com/JHING/ava.demo.datagrid

![GIF 2024-10-8 13-21-30](https://github.com/user-attachments/assets/385cd37e-c186-458c-926c-b858c461e137)

### Avalonia version

11.1.3

### OS

Windows

**Comment by timunie** [comment]
this is by design, as the values will only be updated after lost focus. Use `DataGridTemplateColumn` if you need more control over it.

**Reply by JHING** [reply]
thanks, I have some confusion, I did not modify the value of the added item (id is 6) in viewmodel, and  i also used `DataGridTemplateColumn`, i just removed the first item; when the datagrid has enough height to display the item (id is 6), this problem will not occur.
this is the code in view:
```
 <DataGrid
            BorderBrush="Gray"
            BorderThickness="1"
            GridLinesVisibility="All"
            Height="{Binding DataGridHeight}"
            ItemsSource="{Binding Items}">
            <DataGrid.Columns>
                <DataGridTemplateColumn Header="selected">
                    <DataGridTemplateColumn.CellTemplate>
                        <DataTemplate>
                            <CheckBox
                                Command="{ReflectionBinding #Root.((vm:MainWindowViewModel)DataContext).ItemSelectCommand}"
                                CommandParameter="{Binding Id}"
                                HorizontalAlignment="Center"
                                IsChecked="{ReflectionBinding IsSelected,
                                                              Mode=OneWay}"
                                VerticalAlignment="Center" />
                        </DataTemplate>
                    </DataGridTemplateColumn.CellTemplate>
                </DataGridTemplateColumn>

                <DataGridTextColumn
                    Binding="{Binding Id}"
                    Header="id"
                    IsReadOnly="True" />

            </DataGrid.Columns>
 </DataGrid>
```

and the checkbox  binding command:
```
    [RelayCommand]
    private void ItemSelect(int id)
    {
        var item = Items.FirstOrDefault((DataModel o) => o.Id == id);
        if (item == null)
        {
            return;
        }

        if (SetSelectedBeforeRemove)
        {
            item.IsSelected = false;
        }

        Items.Remove(item);
    }
```
